cmd/dbg/main.go

```diff
@@ -0,0 +1,221 @@
+	generalCmd := &cobra.Command{
+		Use:   "general",
+		Short: "Output the general dynamic lua state",
+		Run: func(cmd *cobra.Command, args []string) {
+			general()
+		},
+	}
+	rootCmd.AddCommand(generalCmd)
+
+	confCmd := &cobra.Command{
+		Use:   "conf",
+		Short: "Dump the contents of /etc/nginx/nginx.conf",
+		Run: func(cmd *cobra.Command, args []string) {
+			readNginxConf()
+		},
+	}
+	rootCmd.AddCommand(confCmd)
+
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func backends() {
+	//Get the backend information from the nginx instance
+	body, requestErr := makeConfigRequest(backendsPath)
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Indent the returned JSON
+	var prettyBuffer bytes.Buffer
+	indentErr := json.Indent(&prettyBuffer, body, "", "  ")
+	if indentErr != nil {
+		fmt.Println(indentErr)
+		return
+	}
+
+	//Print the pretty JSON
+	fmt.Println(string(prettyBuffer.Bytes()))
+}
+
+func backendsList() {
+	body, requestErr := makeConfigRequest(backendsPath)
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Read the array of backends from the returned JSON
+	var f interface{}
+	unmarshalErr := json.Unmarshal(body, &f)
+	if unmarshalErr != nil {
+		fmt.Println(unmarshalErr)
+		return
+	}
+	backends := f.([]interface{})
+
+	//Just list the names of the backends
+	for _, backendi := range backends {
+		backend := backendi.(map[string]interface{})
+		fmt.Println(backend["name"].(string))
+	}
+}
+
+func backendsGet(name string) {
+	body, requestErr := makeConfigRequest(backendsPath)
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Read the array of backends from the returned JSON
+	var f interface{}
+	unmarshalErr := json.Unmarshal(body, &f)
+	if unmarshalErr != nil {
+		fmt.Println(unmarshalErr)
+		return
+	}
+	backends := f.([]interface{})
+
+	//Search for a backend by name and output its config if found
+	for _, backendi := range backends {
+		backend := backendi.(map[string]interface{})
+		if backend["name"].(string) == name {
+			printed, _ := json.MarshalIndent(backend, "", "  ")
+			fmt.Println(string(printed))
+			return
+		}
+	}
+	fmt.Println("A backend of this name was not found.")
+}
+
+func general() {
+	//Get the other (general) information from the nginx instance
+	body, requestErr := makeConfigRequest(generalPath)
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Indent the returned JSON
+	var prettyBuffer bytes.Buffer
+	indentErr := json.Indent(&prettyBuffer, body, "", "  ")
+	if indentErr != nil {
+		fmt.Println(indentErr)
+		return
+	}
+
+	//Print the pretty JSON
+	fmt.Println(string(prettyBuffer.Bytes()))
+}
+
+func readNginxConf() {
+	confFile, err := os.Open("/etc/nginx/nginx.conf")
+	defer confFile.Close()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	contents, _ := ioutil.ReadAll(confFile)
+	fmt.Print(string(contents))
+}
+
+func makeConfigRequest(path string) ([]byte, error) {
```

Comment:
Why not use https://github.com/kubernetes/ingress-nginx/blob/master/internal/nginx/main.go#L51 just to avoid duplications

Reply:
done